client/src/pages/treedata/CarbonCalculator.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable max-len */
+/* eslint-disable react/jsx-one-expression-per-line */
+import React from 'react';
+import DataTable from './DataTable';
+import { format } from 'date-fns';
+
+const carbonCalculatorExplainer = (carbonDetails) => ({
+  weightTotalGreen: `1.2 * weightaboveGround = 1.2 * ${carbonDetails.weightaboveGround} = ${carbonDetails.weightTotalGreen} lbs`,
+  weightDry: `0.725 * weightTotalGreen = 0.725 * ${carbonDetails.weightTotalGreen} = ${carbonDetails.weightDry} lbs`,
+  weightCarbon: ` 0.5 * weightDry = 0.5 * ${carbonDetails.weightDry} = ${carbonDetails.weightCarbon} lbs`,
+  weightCarbonDioxide: `3.67 * weightCarbon  = 3.67 * ${carbonDetails.weightCarbon} = ${carbonDetails.weightCarbonDioxide} lbs CO2 sequestered in ${carbonDetails.age} years`,
+  weightCarbonDioxidePerYear: `weightCarbonDioxide / age  = ${carbonDetails.weightCarbonDioxide} / ${carbonDetails.age} = ${carbonDetails.weightCarbonDioxidePerYear} lbs CO2 sequestered per years`,
+  weightCarbonDioxideInKg: `${carbonDetails.convertWeightCarbonDioxideToKg} kg`,
+});
+
+const carbonCalculatorMath = (age, dbh, height, planted, decimalYear) => ({
+  planted: format(new Date(planted), 'yyyy-MM-dd'),
+  decimalYear,
+  DBH: dbh,
+  height,
+  weightaboveGround: 0.25 * dbh * height,
+  weightTotalGreen: 1.2 * 0.25 * dbh * height,
+  weightDry: 0.725 * 1.2 * 0.25 * dbh * height,
+  weightCarbon: 0.5 * 0.725 * 1.2 * 0.25 * dbh * height,
+  weightCarbonDioxideInLbs: 3.67 * 0.5 * 0.725 * 1.2 * 0.25 * dbh * height,
+  weightCarbonDioxideInKg: 0.45359237 * 3.67 * 0.5 * 0.725 * 1.2 * 0.25 * dbh * height,
+  weightCarbonDioxidePerYearInLbs: (3.67 * 0.5 * 0.725 * 1.2 * 0.25 * dbh * height) / age,
+  weightCarbonDioxidePerYearInKg: (0.45359237 * 3.67 * 0.5 * 0.725 * 1.2 * 0.25 * dbh * height) / age,
+});
+
+const calculateAge = (planted) => {
+  const functionName = 'calculateAge';
+  const dateFormatted = new Date(planted);
```

Comment:
This isn't a string-formatted date, right?  Probably simplest to do `Date.now() - new Date(planted);` on the next line.